### PR TITLE
Fix N+1 queries when rehydrating API rows in AjaxTemplateRenderController

### LINKED ISSUE #241: [Performance/Medium] N+1 queries in AjaxTemplateRenderController::resolveSourceRow
## Severity
**Medium** — N+1.

## Location
`src/Controller/AjaxTemplateRenderController.php:80-92` (`resolveSourceRow`)

## Problem
For each row of the client-supplied batch, a single `->find($id)` is executed:
```php
return $this->doctrine->getRepository($entityClass)->find($id);
```
A page of 100 rows triggers 100 queries.

## Status
**Confirmed.**

## Proposed fix
Collect all identifiers, run a single `findBy(['id' => $ids])`, and re-index the results by id.

## Verification
- Test counting queries (or asserting `findBy` is called once via a mock).

### PR DESCRIPTION
**Pull request by pentiminax**
## Context

Closes #241.

`AjaxTemplateRenderController` re-hydrates API-supplied rows into Doctrine entities so that template, URL, and action columns can be resolved server-side. Previously `resolveSourceRow()` ran a single `getRepository($entityClass)->find($id)` **per row**, so a page of 100 rows issued 100 SELECT queries — a classic N+1 (confirmed, Medium severity).

## Changes

- **Batch the lookup into one query.** All identifiers are collected up front and resolved with a single `findBy([$idField => $ids])`, then re-indexed by identifier.
- **Keep the controller free of business logic (SOLID / clean code).** The Doctrine query and identifier-parsing logic moved out of the controller into three single-responsibility services under a new `Pentiminax\UX\DataTables\Rehydration` namespace, mirroring existing collaborator services like `EntityLocator`:
  - `RowIdentifierExtractor` — parses the identifier (`id` / `@id` IRI) from a client row.
  - `SourceRowResolver` — batch-rehydrates entities in a single query. Uses Doctrine metadata (`getIdentifierFieldNames()` / `getIdentifierValues()`, consistent with `EditFormService` and `EntityLocator`), so it also handles entities whose identifier field is not literally `id`.
  - `SourceRowMap` — value object answering "which entity backs this row?", so the controller never touches identifiers.
- `AjaxTemplateRenderController` now drops its `ManagerRegistry` dependency and is pure orchestration.
- DI wiring added in `config/services.php` (explicit style matching the file).

## Tests

- `tests/Unit/Rehydration/SourceRowResolverTest.php` — core regression: a multi-row batch triggers **exactly one** `findBy` call with all ids, and the map resolves each row; plus empty-map paths (no entity class, no doctrine, no resolvable id).
- `tests/Unit/Rehydration/RowIdentifierExtractorTest.php` — `id`, `@id` IRI, and null/empty cases.
- `tests/Unit/Controller/AjaxTemplateRenderControllerTest.php` — updated to inject the real `SourceRowResolver`; existing rendered-output assertions unchanged.

Full suite: **814 tests, 2147 assertions — green**. Code style verified with PHP-CS-Fixer.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01YWmxpPz3JaDwQHKL2ZvUyq

---
_Generated by [Claude Code](https://claude.ai/code/session_01YWmxpPz3JaDwQHKL2ZvUyq)_

---

> [!NOTE]
> **Low Risk**
> Scoped refactor of AJAX row rehydration with preserved fallbacks and regression tests; no auth or mutation path changes.
> 
> **Overview**
> Fixes an **N+1** on the AJAX template-render endpoint by resolving Doctrine entities for all submitted rows in **one** `findBy` instead of per-row `find`.
> 
> Rehydration moves out of `AjaxTemplateRenderController` into `RowIdentifierExtractor` (reads `id` or `@id` IRI) and `SourceRowResolver` (batch lookup via metadata-driven identifier fields, aligned with `EditFormService` / `EntityLocator`). The controller now depends on `SourceRowResolver` only; composite-key entities are left unrehydrated rather than mismatched.
> 
> DI registers the new services and wires them into `datatables.controller.ajax_templates`. Unit tests cover batching, key alignment, edge cases, and updated controller expectations (`findBy` instead of `find`).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 1c6f15f88b811f8abdb163999b2c4ecb5d47d6d5. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/Rehydration/SourceRowMap.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    /**
+     * @param array<string, mixed> $row
+     */
+    public function sourceFor(array $row): ?object
+    {
+        $id = $this->identifierExtractor->extract($row);
+
+        return null !== $id ? ($this->entities[(string) $id] ?? null) : null;
```
[the remark this answers is not in the document]
Comment (pentiminax) [reply src/Rehydration/SourceRowMap.php]: Fixed in 1c6f15f. Rather than pushing identifier extraction back to the controller, `SourceRowResolver::resolve()` now returns the entities aligned to their original row keys, so the identifier is extracted exactly once per row (during the batch build) and the controller just indexes the result. The `SourceRowMap` value object — and its service dependency — is removed entirely.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/Rehydration/SourceRowResolver.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        }
+
+        $manager = $this->doctrine->getManagerForClass($entityClass);
+
+        if (!$manager instanceof ObjectManager) {
+            return SourceRowMap::empty($this->identifierExtractor);
+        }
+
+        $metadata = $manager->getClassMetadata($entityClass);
+        $idField  = $metadata->getIdentifierFieldNames()[0] ?? 'id';
```
[the remark this answers is not in the document]
Comment (pentiminax) [reply src/Rehydration/SourceRowResolver.php]: Fixed in 1c6f15f. `resolve()` now guards on `count(getIdentifierFieldNames()) !== 1` and skips rehydration for composite-key entities (returning all-null, so rows fall back to their raw client data) instead of querying by a single PK field and silently dropping colliding entities. Added a test covering this (`it_skips_rehydration_for_composite_key_entities`).

---
_Generated by [Claude Code](https://claude.ai/code)_
